### PR TITLE
Sg/endpoints

### PR DESCRIPTION
**Pull request by sankethgunasekara**
Have added the Endpoints for the Input requirement request and have Retrieve the data and showed in the Table

### CODE REVIEW

File: RFPAIPCF/src/App.tsx
Comment (tilangapramithvidusara) [inline RFPAIPCF/src/App.tsx:1]: change this class component as functional component

File: RFPAIPCF/src/components/Table.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
+import React, { useState, useEffect } from "react";
+import "../styles/tableStyles.css";
+import { Modal, Button, Input } from "antd";
+import { PlusOutlined, CloseCircleOutlined } from "@ant-design/icons";
+
+interface TableProps {
```
Comment (tilangapramithvidusara) [inline RFPAIPCF/src/components/Table.tsx]: move this interfaces into common types file

File: RFPAIPCF/src/components/Table.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      delete updatedMap[id];
+      return updatedMap;
+    });
+  }
+  return (
+    <>
+      <table className="custom-table">
+        <thead>
+          <tr>
+            <th>REQUIREMENTS</th>
```
Comment (tilangapramithvidusara) [inline RFPAIPCF/src/components/Table.tsx]: Get this hard coded view Text from language json

File: RFPAIPCF/src/components/Table.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            <th>SEER RFP RESPONSE</th>
+          </tr>
+        </thead>
+        <tbody>
+          {data.map((row, index) => (
+            <tr
+              key={index}
+              className={index % 2 === 0 ? "even-row" : "odd-row"}
+            >
+              <td>{row.requirement}</td>
```
Comment (tilangapramithvidusara) [inline RFPAIPCF/src/components/Table.tsx]: use ? operator

File: RFPAIPCF/src/components/Table.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+                  onClick={handlePopUpOpen}
+                />
+              </td>
+              <td>{row.seerRFPResponse}</td>
+            </tr>
+          ))}
+        </tbody>
+      </table>
+      <Modal
+        title="Enter Your Text"
```
Comment (tilangapramithvidusara) [inline RFPAIPCF/src/components/Table.tsx]: get from language json

File: RFPAIPCF/src/components/Table.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+        open={isModalOpen}
+        onOk={handlePopUpSubmit}
+        onCancel={handlePopUpCancel}
+      >
+        <Input.TextArea
+          className="modalTextArea"
+          rows={6}
+          placeholder="Enter Your Text Here..."
```
Comment (tilangapramithvidusara) [inline RFPAIPCF/src/components/Table.tsx]: get from language json

File: RFPAIPCF/src/pages/UserInput.tsx [hunk 6]
```diff
@@ -1,9 +1,77 @@
...
+          const seerRFPResponse = "Seer RFP Response 1";
+          const newData = [
+            { id, requirement, seerRequirement, seerRFPResponse },
+          ];
+          setData(newData);
+        } else {
+          console.error(
+            "Invalid response format: requirement property is missing"
+          );
+          message.error("Invalid response format");
```
Comment (tilangapramithvidusara) [inline RFPAIPCF/src/pages/UserInput.tsx]: get from language json

**Review by tilangapramithvidusara — CHANGES_REQUESTED** [review]

**Comment by sankethgunasekara** [comment]
I have done the requested changes

**Review by tilangapramithvidusara — APPROVED** [review]

**Pull request merged**